libs/shared/flagd-js-core/src/lib/flagd-js-core.ts

```diff
@@ -0,0 +1,133 @@
+import { Storage, StorageImpl } from './storage';
+import {
+  ErrorCode,
+  EvaluationContext,
+  FlagValue,
+  JsonValue,
+  ResolutionDetails,
+  StandardResolutionReasons,
+} from '@openfeature/server-sdk';
+
+/**
+ * Expose flag configuration setter and flag resolving methods.
+ */
+export class FlagdJSCore {
+  private _storage: Storage;
+
+  /**
+   * Construct with optional your own storage layer.
+   */
+  constructor(storage?: Storage) {
+    this._storage = storage ? storage : new StorageImpl();
+  }
+
+  setConfigurations(cfg: string): void {
+    this._storage.setConfigurations(cfg);
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    transformedContext: EvaluationContext,
+  ): ResolutionDetails<boolean> {
+    return this.resolve(flagKey, defaultValue, transformedContext, (v) => {
+      return typeof v == 'boolean';
+    });
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    transformedContext: EvaluationContext,
+  ): ResolutionDetails<string> {
+    return this.resolve(flagKey, defaultValue, transformedContext, (v) => {
+      return typeof v == 'string';
+    });
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    transformedContext: EvaluationContext,
+  ): ResolutionDetails<number> {
+    return this.resolve(flagKey, defaultValue, transformedContext, (v) => {
+      return typeof v == 'number';
+    });
+  }
+
+  resolveObjectEvaluation<T extends JsonValue>(
+    flagKey: string,
+    defaultValue: T,
+    transformedContext: EvaluationContext,
+  ): ResolutionDetails<T> {
+    return this.resolve(flagKey, defaultValue, transformedContext, (v) => {
+      return typeof v == 'object';
+    });
+  }
+
+  private resolve<T extends FlagValue>(
+    flagKey: string,
+    defaultValue: T,
+    transformedContext: EvaluationContext,
+    guard: (a: FlagValue) => boolean,
+  ): ResolutionDetails<T> {
+    // flag exist
+    const flag = this._storage.getFlag(flagKey);
+    if (flag === undefined) {
+      return {
+        value: defaultValue,
+        errorCode: ErrorCode.FLAG_NOT_FOUND,
+        errorMessage: `flag: ${flagKey} not found`,
+        reason: StandardResolutionReasons.ERROR,
+      };
+    }
+
+    // flag status
+    if (flag.state == 'DISABLED') {
+      return {
+        value: defaultValue,
+        errorCode: ErrorCode.GENERAL,
```

Comment:
Should this be a `FLAG_NOT_FOUND` error?

Reply:
Introrudced errors. Updated in the upstream PR -  https://github.com/open-feature/js-sdk-contrib/pull/620